job-server/src/main/scala/db/h2/migration/V0_7_2/V0_7_2__Migrate_Blobs.scala

```diff
@@ -0,0 +1,56 @@
+package db.h2.migration.V0_7_2
+
+import java.sql.Connection
+
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.TimeoutException
+import scala.concurrent.duration.DurationInt
+
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration
+import org.slf4j.LoggerFactory
+
+import javax.sql.rowset.serial.SerialBlob
+import slick.dbio.DBIO
+import slick.driver.H2Driver.api.actionBasedSQLInterpolation
+import slick.jdbc.GetResult
+import slick.jdbc.PositionedParameters
+import slick.jdbc.SetParameter
+import spark.jobserver.slick.unmanaged.UnmanagedDatabase
+
+class V0_7_2__Migrate_Blobs extends JdbcMigration {
+  private val Timeout = 10 minutes
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  private case class BinArray(id: Int, binary: Array[Byte])
+
+  private def logErrors = PartialFunction[Throwable, Unit] {
+    case e: Throwable => logger.error(e.getMessage, e)
+  }
+
+  private def insertBlob(db: UnmanagedDatabase, b: BinArray): Unit = {
+    implicit object SetSerialBlob extends SetParameter[SerialBlob] {
+      def apply(v: SerialBlob, pp: PositionedParameters) {
+        pp.setBlob(v)
+      }
+    }
+    val blob = new SerialBlob(b.binary)
+    val updateBlob = sqlu"""UPDATE BINARIES SET BINARY=${blob} WHERE BIN_ID=${b.id}"""
+    Await.ready(db.run(updateBlob).recover{logErrors}, Timeout)
+  }
+
+  def migrate(c: Connection): Unit = {
+    val renameBinaryToBinArray = sqlu"""ALTER TABLE BINARIES ALTER COLUMN BINARY RENAME TO BINARRAY"""
+    val addBlobColumn = sqlu"""ALTER TABLE BINARIES ADD COLUMN BINARY BLOB"""
+    val dropBinArray = sqlu"""ALTER TABLE BINARIES DROP COLUMN BINARRAY"""
+    implicit val getBinArrayResult = GetResult[BinArray](r => BinArray(r.nextInt(), r.nextBytes()))
+    val getBinArrays = sql"""SELECT BIN_ID, BINARRAY FROM BINARIES""".as[BinArray]
+
+    val db = new UnmanagedDatabase(c)
+    c.setAutoCommit(false)
+    Await.ready(db.run(DBIO.seq(renameBinaryToBinArray, addBlobColumn)).recover{logErrors}, Timeout)
```

Comment:
You could use a for comprehension to sequence all these Futures from db.run, then just use a single Await.ready and recover.